src/Microsoft.DotNet.Darc/DarcLib/VirtualMonoRepo/VmrUpdater.cs

```diff
@@ -182,10 +172,34 @@ public class VmrUpdater : VmrManagerBase, IVmrUpdater
         bool discardPatches,
         CancellationToken cancellationToken)
     {
-        var currentSha = GetCurrentVersion(update.Mapping);
+        VmrDependencyVersion currentVersion = _dependencyTracker.GetDependencyVersion(update.Mapping)
+            ?? throw new Exception($"Failed to find current version for {update.Mapping.Name}");
+
+        _logger.LogInformation("Synchronizing {name} from {current} to {repo} / {revision}",
+            update.Mapping.Name,
+            currentVersion.Sha,
+            update.RemoteUri,
+            update.TargetRevision);
 
-        _logger.LogInformation("Synchronizing {name} from {current} to {repo} / {revision}{oneByOne}",
-            update.Mapping.Name, currentSha, update.RemoteUri, update.TargetRevision, noSquash ? " one commit at a time" : string.Empty);
+        // Do we need to change anything?
+        if (currentVersion.Sha == update.TargetRevision)
+        {
+            if (currentVersion.PackageVersion != update.TargetVersion && update.TargetVersion != null)
```

Comment:
nit: should the order of these conditions be swapped? Think the compiler will make it so we don't look at the second one if `update.TargetVersion == null` in that case

Reply:
Yeah but actually the first one will be false more often than the second so this micro-optimization doesn't make sense even in that regard